esphome/components/font/__init__.py

```diff
@@ -154,31 +182,44 @@ def validate_glyphs(config):
             points = {ord(x) for x in flatten(extra[CONF_GLYPHS])}
             glyphspoints.difference_update(points)
             setpoints.difference_update(points)
-            check_missing_glyphs(extra[CONF_FILE][CONF_PATH], points)
+            check_missing_glyphs(extra[CONF_FILE], points)
 
         # A named glyph that can't be provided is an error
-        check_missing_glyphs(fileconf[CONF_PATH], glyphspoints)
+
+        check_missing_glyphs(fileconf, glyphspoints)
         # A missing glyph from a set is a warning.
         if not config[CONF_IGNORE_MISSING_GLYPHS]:
-            check_missing_glyphs(fileconf[CONF_PATH], setpoints, warning=True)
+            check_missing_glyphs(fileconf, setpoints, warning=True)
 
     # Populate the default after the above checks so that use of the default doesn't trigger errors
     if not config[CONF_GLYPHS] and not config[CONF_GLYPHSETS]:
         if fileconf[CONF_TYPE] == TYPE_LOCAL_BITMAP:
             config[CONF_GLYPHS] = [DEFAULT_GLYPHS]
         else:
             # set a default glyphset, intersected with what the font actually offers
-            font = FONT_CACHE[fileconf[CONF_PATH]]
+            font = FONT_CACHE[fileconf]
             config[CONF_GLYPHS] = [
                 chr(x)
                 for x in glyphsets.unicodes_per_glyphset(DEFAULT_GLYPHSET)
                 if font.get_char_index(x) != 0
             ]
 
+    if config[CONF_FILE][CONF_TYPE] == TYPE_LOCAL_BITMAP and CONF_SIZE in config:
+        raise cv.Invalid(
+            "Size is not a valid option for bitmap fonts, which are inherently fixed size"
+        )
```

Comment:
We should set the default on the config if it is not a bitmap and not defined by the user.
This makes it more clear with the config validation/dump as it will show up there and basically acts the same way as the original `default=20`.

```suggestion
    if config[CONF_FILE][CONF_TYPE] == TYPE_LOCAL_BITMAP:
        if CONF_SIZE in config:
            raise cv.Invalid(
                "Size is not a valid option for bitmap fonts, which are inherently fixed size"
            )
    elif CONF_SIZE not in config:
        config[CONF_SIZE] = 20
```

Reply:
> We should set the default on the config if it is not a bitmap and not defined by the user

Agree, but it's complicated :-) I'll have a look at how that can be done without jumping through too many hoops.